plugins/loop_registry.go

```diff
@@ -45,14 +42,15 @@ func NewLoopRegistry(lggr logger.Logger, tracing config.Tracing, telemetry confi
 	}
 }
 
-// Register creates a port of the plugin. It is not idempotent. Duplicate calls to Register will return [ErrExists]
+// Register creates a port of the plugin. It is idempotent. Duplicate calls to Register will return the same port.
 // Safe for concurrent use.
 func (m *LoopRegistry) Register(id string) (*RegisteredLoop, error) {
 	m.mu.Lock()
 	defer m.mu.Unlock()
 
-	if _, exists := m.registry[id]; exists {
-		return nil, ErrExists
+	if l, exists := m.registry[id]; exists {
+		m.lggr.Warnf("Trying to register a loop that is already registered %q", id)
```

Comment:
Hmmmm -- is this all we need to do? I would have thought that there would now be multiple loopps running, one of which is orhpaned because I don't see us cleaning it up.

Reply:
i believe it was a little different.

we are passing the [loop server back](https://github.com/smartcontractkit/chainlink/blob/0e0d19e719a7e5905c52e28009ba4406409640ce/core/services/ocr2/plugins/mercury/plugin.go#L262)as one of the [job ServiceCtx objects](https://github.com/smartcontractkit/chainlink/blob/0e0d19e719a7e5905c52e28009ba4406409640ce/core/services/job/spawner.go#L210) and that will be closed when the active job [closes all it's subservices](https://github.com/smartcontractkit/chainlink/blob/0e0d19e719a7e5905c52e28009ba4406409640ce/core/services/job/spawner.go#L158)


rather, we are forgetting to unregister the loop as part of this job close'ing

i refactored to make a much more targeted fix that only applies to mercury and a follow up ticket to make the pattern reusable for other jobs.

note that the [generic jobs are already unregistering](https://github.com/smartcontractkit/chainlink/blob/0e0d19e719a7e5905c52e28009ba4406409640ce/core/services/ocr2/validate/validate.go#L282)and this should only effect job that really on custom providers